### ISSUE TITLE
uploading a non-ODD results in blank screen of death

### ISSUE THREAD

**Issue by sydb** [TEIC/romajs#164]
If you 

- go to https://roma.tei-c.org/
- click on “UPLOAD ODD”
- click on the tiny “Browse…” button
- select an XML file that is not a “proper” TEI ODD (see below)
- click “START”
- URL in address bar changed to https://roma.tei-c.org/settings, but …

The resulting screen is entirely blank. Looking at source of the blank web page the `<html:body>` has an empty `<html:div id=romajs>` and an empty `<html:script src=…>`.

To be fair, I do not expect that Roma _do_ much with something that does not meet its criteria for a proper ODD file. But it should at least give a “I’m sorry, I cannot deal with this file” error message. I consider this a BUG, thus the “bug” label for this ticket.

**below**
I do not know what criteria Roma uses to define a proper ODD file. I tried a DocBook document, and not surprisingly it did not work. I tried an ODD which had nothing but a `<schemaSpec>`, and that did not work, either. It would be nice if Roma further differentiated _why_ an uploaded file is not usable. Something like:

- Not an XML file — when ill-formed
- Not a TEI document — when outermost element is not in TEI namespace
- Missing required XXXX — when `<title>` (or anything else required) is missing from `<teiHeader>` (or, indeed, whole `<teiHeader>` is missing, of course)
- Not a TEI ODD customization — when `<schemaSpec>` is missing
- Invalid TEI ODD customization — when invalid against, say, tei_odds.

I consider this a good idea, but much less important, thus an enhancement; hence the “enhancement” label, too.

**Comment by raffazizzi** [comment]
@sydb can you tell me more about the browser and environment you're using? When trying to reproduce this, I get the expected behavior of a modal letting you know you're doing something unexpected:

![image](https://github.com/TEIC/romajs/assets/144770/e9058756-3f9b-4d9c-8835-3872e5ca2801)

**Comment by raffazizzi** [comment]
Nevermind: I could reproduce this on roma.tei-c.org, but it's puzzling. The error pop up works correctly both locally on a Netlify dev instance. Could it be something with how Roma is bootstrapped on tei-c.org? Or maybe this would simply be fixed by making a new release?

**Comment by peterstadler** [comment]
There were no automated pipelines in place to build the Docker images at https://hub.docker.com/r/teic/romajs.
Both dev and latest were more than a year old.

I just added a GitHub action ("[Docker](https://github.com/TEIC/romajs/actions/workflows/docker-publish.yml)") to build and push the Docker images for every git push. It already built successfully and the dev image is updated now. This updated dev image is deployed at https://roma.tei-c.de/ (at Paderborn University) and shows the expected behavior in catching the error and displaying the modal.

The deployment at https://roma.tei-c.org/ (at Huma-Num) is running from the old latest image. How should we proceed there? Options I see

1. do a manual update of `teic/romajs:latest` from current master branch
2. do an empty commit to the master branch. This will trigger the GitHub pipeline and automatically update the Docker image `teic/romajs:latest`
3. do a point(?) release by merging current dev into master branch. This will also trigger the GitHub pipeline and automatically update the Docker image `teic/romajs:latest`

The deployment at Huma-Num needs to be updated manually afterwards in any case.
